### DISCUSSION TITLE
Draw frame around the table

### DISCUSSION THREAD

**Discussion by mabana** [prawnpdf/prawn#1326] [General]
Hello,

I've been trying to create a report with several sections, some of them are tables(prawn-table), but not all of them. Each of section should be in a frame.

I have two problems which I'm not able to resolve.
Those problems arise when table is to big to draw it on one page.
First problem is that prawn draw second part of the table(on the second page) from middle of the page but i need to draw this second part from top of the page.

Second problem is that i can't draw frame around table when table is splitted to few pages.

```
require 'prawn'
require 'prawn/table'

class Report
  include Prawn::View

  def render(data)
    border do
      text '1'
    end

    border do
      text '2' * 100
    end

    border do
      table(data, width: bounds.width)
    end
  end

  private

  def border
    padding = 30

    bounding_box([0, cursor], width: bounds.width) do
      move_down padding

      bounding_box([padding, cursor], width: bounds.width - (padding * 2)) do
        yield
      end

      move_down padding

      stroke_bounds
    end

    move_down 20
  end
end

Report.new.tap do |report|
  data = Array.new(100, [1, 2, 3, 4])
  report.render(data)
end.save_as("report.pdf")

```
[report.pdf](https://github.com/prawnpdf/prawn/files/2175332/report.pdf)

Do you have any idea how to achieve it?

**Comment by pointlessone** [comment]
You probably can't draw a frame around a table split between pages because `stroke_bounds` doesn't handle page breaks. Technically, bounding boxes do not span across page breaks. Those are two different bounding boxes on different pages. So that last call strokes whatever happens to be the current bounding box, which is on the last page.

The issue with the second part of the table being drawn in the middle of the page is the `bounding_box`. The second page inherits bounds from the first one. This is how bounding boxes work. In order to draw frames you probably better don't use them at all. Instead remember the cursor position before your content, then drawn a rectangle from the new cursor position to the old one. It's probably more complicated if you also want to handle page breaks.
